File: app/src/main/java/databinding/example/com/java/data/ProfileObservableViewModel.java
package databinding.example.com.java.data;

import android.databinding.Bindable;
import android.databinding.ObservableField;
import android.databinding.ObservableInt;

import databinding.example.com.java.util.ObservableViewModel;

public class ProfileObservableViewModel extends ObservableViewModel {

    public void increment() {
        likes.set(likes.get()+1);
    }

    public ObservableField name = new ObservableField("Ada333");
    public ObservableField lastName = new ObservableField("Lovelace");
    public ObservableInt likes = new ObservableInt(0);

    public void onLike() {
        increment();
    }

    @Bindable
    public Popularity getPopularity() {
        Popularity result;
        if (likes.get() > 9){
            result = Popularity.STAR;
        }else if(likes.get() > 4 && likes.get() <= 9){
            result = Popularity.POPULAR;
        }else{
            result = Popularity.NORMAL;
        }

        return result;
    }
}

File: app/src/main/java/databinding/example/com/basic/data/ProfileObservableViewModel.kt
package databinding.example.com.basic.data

import android.arch.lifecycle.ViewModel
import android.databinding.Bindable
import android.databinding.ObservableField
import android.databinding.ObservableInt
import databinding.example.com.BR
import databinding.example.com.basic.util.ObservableViewModel

class ProfileObservableViewModel : ObservableViewModel() {
    val name = ObservableField("Ada333")
    val lastName = ObservableField("Lovelace")
    val likes =  ObservableInt(0)

    fun onLike() {
        likes.increment()
        notifyPropertyChanged(BR.popularity)
    }

    @Bindable
    fun getPopularity(): Popularity {
        return likes.get().let {
            when {
                it > 9 -> Popularity.STAR
                it > 4 -> Popularity.POPULAR
                else -> Popularity.NORMAL
            }
        }
    }
}

class ProfileObservableFieldsViewModel : ViewModel() {
    val name = ObservableField("Ada22")
    val lastName = ObservableField("Lovelace")
    val likes =  ObservableInt(0)

    val popularity = ObservableField<Popularity>(Popularity.NORMAL)

    fun onLike() {
        likes.set(likes.get() + 1)

        popularity.set(likes.get().let {
            if (it > 9) Popularity.STAR
            if (it > 4) Popularity.POPULAR
            Popularity.NORMAL
        })
    }
}

enum class Popularity {
    NORMAL,
    POPULAR,
    STAR
}

private fun ObservableInt.increment() {
    set(get() + 1)
}